middleware/grpc/interceptor.go

```diff
@@ -65,22 +71,75 @@ type (
 // The new middleware is created with default identity and policy path mapper.
 // Those can be overridden using `Middleware.Identity` to specify the caller's identity, or using
 // the middleware's ".With...()" functions to set policy path and resource mappers.
-func New(client AuthorizerClient, policy Policy) *Middleware {
+func New(authzClient AuthorizerClient, policy Policy) *Middleware {
 	policyMapper := methodPolicyMapper("")
 	if policy.Path != "" {
 		policyMapper = nil
 	}
 
 	return &Middleware{
-		client:          client,
+		client:          authzClient,
 		Identity:        (&IdentityBuilder{}).FromMetadata("authorization"),
 		policyContext:   *internal.DefaultPolicyContext(policy),
 		policyInstance:  *internal.DefaultPolicyInstance(policy),
 		policyMapper:    policyMapper,
 		resourceMappers: []ResourceMapper{},
+		ignoredPaths:    []string{},
 	}
 }
 
+func NewFromConfig(ctx context.Context, cfg *AuthorizationConfig) (*Middleware, error) {
+	if cfg == nil {
+		return nil, errors.New("config wasn't provided")
+	}
+
+	policyMapper := methodPolicyMapper("")
+
+	if cfg.Policy.Path != "" {
+		policyMapper = nil
+	}
+
+	dialOptionsProvider := grpcclient.NewDialOptionsProvider()
+
+	authzClient, err := newAuthorizerClient(ctx, dialOptionsProvider, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Middleware{
+		client:          authzClient,
+		Identity:        (&IdentityBuilder{}).FromMetadata("authorization"),
+		policyContext:   *internal.DefaultPolicyContext(cfg.Policy),
+		policyInstance:  *internal.DefaultPolicyInstance(cfg.Policy),
+		policyMapper:    policyMapper,
+		resourceMappers: []ResourceMapper{},
+		cfg:             cfg,
+		ignoredPaths:    []string{},
+	}, nil
+}
+
+func newAuthorizerClient(ctx context.Context,
+	dop grpcclient.DialOptionsProvider,
+	cfg *AuthorizationConfig,
+) (AuthorizerClient, error) {
+	connOpts, err := cfg.ToClientOptions(dop)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to convert client config")
+	}
+
+	conn, err := client.NewConnection(ctx, connOpts...)
+	if err != nil {
+		return nil, errors.Wrap(err, "create grpc client failed")
+	}
+
+	return authz.NewAuthorizerClient(conn.Conn), nil
+}
+
+func (m *Middleware) WithIgnoredPolicyPaths(paths []string) *Middleware {
```

Comment:
I'd rename this to `WithIgnoredMethods`. The thing that is being ignored here is a gRPC method, not a policy path. If a method is ignored then the policy doesn't even need to include a package with that path.